omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/VisitConfigurationController2_0.java

```diff
@@ -0,0 +1,180 @@
+		AdministrationService administrationService = Context.getAdministrationService();
+		EncounterService encounterService = Context.getEncounterService();
+		VisitService visitService = Context.getVisitService();
+		SchedulerService schedulerService = Context.getSchedulerService();
+
+		VisitConfiguration visitConfiguration = new VisitConfiguration();
+		visitConfiguration.setEnableVisits(getEnableVisitsValue(administrationService));
+		visitConfiguration
+				.setEncounterVisitsAssignmentHandler(getVisitEncounterHandlerValue(administrationService, encounterService));
+		visitConfiguration.setStartAutoCloseVisitsTask(getAutoCloseVisitsTaskStartedValue(schedulerService));
+		visitConfiguration.setVisitTypesToAutoClose(getVisitTypesToAutoCloseValue(administrationService, visitService));
+
+		return ConversionUtil.convertToRepresentation(visitConfiguration, Representation.FULL);
+	}
+
+	@RequestMapping(method = RequestMethod.POST)
+	@ResponseStatus(HttpStatus.OK)
+	public void updateCurrentConfiguration(@RequestBody VisitConfiguration newConfiguration) throws SchedulerException {
+		Context.requirePrivilege(PrivilegeConstants.CONFIGURE_VISITS);
+		AdministrationService administrationService = Context.getAdministrationService();
+		VisitService visitService = Context.getVisitService();
+		SchedulerService schedulerService = Context.getSchedulerService();
+
+		// validate
+		if (newConfiguration.getEnableVisits() && StringUtils.isEmpty(newConfiguration.getEncounterVisitsAssignmentHandler())) {
+			throw new IllegalRequestException("Encounter Visit assignment handler cannot be empty");
+		}
+
+		updateEnableVisitsValue(administrationService, newConfiguration.getEnableVisits());
+
+		if (newConfiguration.getEnableVisits()) {
+			administrationService.setGlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER,
+					newConfiguration.getEncounterVisitsAssignmentHandler());
+		}
+		updateGetAutoCloseVisitsTaskStartedValue(schedulerService, newConfiguration.getStartAutoCloseVisitsTask());
+		updateVisitTypesToAutoCloseValue(administrationService, visitService, newConfiguration.getVisitTypesToAutoClose());
+	}
+
+	private Boolean getEnableVisitsValue(AdministrationService administrationService) {
+		String enableVisits = administrationService
+				.getGlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS, Boolean.FALSE.toString());
+		return Boolean.parseBoolean(enableVisits);
+	}
+
+	private void updateEnableVisitsValue(AdministrationService administrationService, Boolean value) {
+		GlobalProperty enableVisitsGp = new GlobalProperty(OpenmrsConstants.GLOBAL_PROPERTY_ENABLE_VISITS,
+				Boolean.toString(value));
+		administrationService.saveGlobalProperty(enableVisitsGp);
+	}
+
+	private String getVisitEncounterHandlerValue(AdministrationService administrationService,
+			EncounterService encounterService) {
+		String visitEncounterHandler = administrationService.getGlobalProperty(OpenmrsConstants.GP_VISIT_ASSIGNMENT_HANDLER);
```

Comment:
Shouldn't be enough to just return visitEncounterHandler?

Reply:
This method will return null if set visitEncounterHandler is not a valid class name (not existing on invalid). If we don't need that validation then it could just return visitEncounterHandler

Reply:
Are validations or get or save?

Reply:
I've just moved the validation to be on save and removed it from on get. https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/4e041a63b98f21cc6c0ad5dccb14cc235b943c05